test/suites/integration/ami_test.go

```diff
@@ -314,3 +330,19 @@ func getInstanceAttribute(nodeName string, attribute string) *ec2.DescribeInstan
 	Expect(err).ToNot(HaveOccurred())
 	return instanceAttribute
 }
+
+func EventuallyExpectAMIs(provider *v1alpha1.AWSNodeTemplate, amiID string) {
+	Eventually(func(g Gomega) {
+		var ant v1alpha1.AWSNodeTemplate
+		g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(provider), &ant)).To(Succeed())
+		amiIDsInStatus := lo.Map(ant.Status.AMIs, func(ami v1alpha1.AMI, _ int) string {
+			return ami.ID
+		})
+		g.Expect(amiIDsInStatus).To(Equal([]string{amiID}))
+		g.Expect(ant.Status.AMIs[0].Requirements[:1:1]).To(Equal(
```

Comment:
This seems like an odd way to ensure requirements. Why not just use a Contains() check for the requirements set against a static requirement?

Reply:
This is just to limit the capacity of the. This it just limit the capacity during comparison.